File: 0035-search-insert-position/0035-search-insert-position.py
class Solution(object):
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        n = len(nums)
        for i in range(0,n):
            if nums[i] == target:
                return i
            elif ((nums[i] > target) and (nums[i - 1] < target)):
                return i
            elif (nums[i] < target) and i == n-1:
                return i + 1
            elif target < nums[0]:
                return 0

File: 0035-search-insert-position/0035-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int n = nums.length;
        int low = 0;
        int high = n - 1;
        int lb = n;
        while(low <= high){
            int mid = (low + high)/2;
            if(target <= nums[mid]){
                lb = mid;
                high = mid - 1;
            }
            else if(target == nums[mid]){
                return mid;
            }
            else{
                low = mid + 1;
            }
        }   
        return lb;
    }
}